### ISSUE TITLE
Extensions: WP Super Cache settings tweaks

### ISSUE THREAD

**Issue by rickybanister** [Automattic/wp-calypso#12885]
The Super Cache extension is pretty awesome, there's a ton in there and it's been very well translated to Calypso's components. I scrubbed through and documented some of the less standard things:

**Easy tab**

- [x] The main settings card contains text prefaced 'notice'. We should use the `notice` component instead. Something like this:
![image](https://cloud.githubusercontent.com/assets/437258/24773707/38906878-1ae4-11e7-8ede-0ed6264be36f.png)

- [x] The primary toggle 'caching on (recommended)' is confusing. Our toggles labels tend to want to be written as the result of the toggle being on. The toggle then implies the 'on-ness'. So a label more like `Speed up your site with simple caching` or something would work better. The (recommended) bit doesn't seem to help anything. 

- [ ] The `test cache` and `delete cache` buttons don't provide any feedback. Can we trigger a global success notice when the cache is validated or deleted?

**Advanced tab**

- [x] Advanced options: can the sub-option radio buttons be indented? They should perhaps also be disabled if the main caching toggle is off. It would also be nice to avoid the need for the save button and save the form onChange since there are no input fields. 

So this:
![image](https://cloud.githubusercontent.com/assets/437258/24773851/cac3b5d8-1ae4-11e7-8395-b3a7ed273f55.png)

Would become this:
![image](https://cloud.githubusercontent.com/assets/437258/24773991/3cb814d6-1ae5-11e7-92ce-454b964eabdd.png)

- [x] Miscellaneous: The warning text is easy to miss. I did a mockup of using a compact notice for this instead of the description text. I found it a little confounding that we 'recommend' that feature, but then also warn. It might be better to have backend validation to prevent an unsafe combination rather than put the burden on the user reading these settings.
![image](https://cloud.githubusercontent.com/assets/437258/24774106/b054e590-1ae5-11e7-8a24-a3d713fde029.png)

- [x] Secret key: what is this for? There are no instructions and it doesn't seem to relate to anything nearby. Can there be a preamble for this? Should it be in its own card?
![image](https://cloud.githubusercontent.com/assets/437258/24774148/e02f5048-1ae5-11e7-9fba-22785da6d118.png)

Accepted / Rejected stuff: I think we should use a short bold field label for these and restrict use of the italic description for further clarification. When looking at this I couldn't tell if the label was above or below the field:
This:
![image](https://cloud.githubusercontent.com/assets/437258/24774222/21d74708-1ae6-11e7-84e3-d4f8bbda0c93.png)
Would be more like:
![image](https://cloud.githubusercontent.com/assets/437258/24774427/d91ba6e8-1ae6-11e7-8239-67aa556dc106.png)

- [x] Lockdown: can we rewrite this a bit to put a toggle at the very beginning and have the copy say something more like the second screenshot below. The nitty-gritty details below the toggle would become italic description text.
This:
![image](https://cloud.githubusercontent.com/assets/437258/24774465/ff34e54c-1ae6-11e7-902a-0821e455aeb7.png)
To this: 
![image](https://cloud.githubusercontent.com/assets/437258/24774506/2fa3d8a0-1ae7-11e7-8753-182909cd1cc3.png)

**CDN tab**

This looks nearly perfect. Everything is very clean. My only recommendations are:

- [x] Indent everything beneath the main toggle 40px over so there is less confusion when CDN caching is not on
- [x] If the feature is toggled off should the fields be disabled or editable? I realize someone might want to configure things before enabling the feature, but I could also see a case for disabling all the fields if CDN caching is off.
- [ ] I see your url as the default text in the url field:
![image](https://cloud.githubusercontent.com/assets/437258/24774633/b514d048-1ae7-11e7-9536-c1228575b858.png)

cc @beaulebens @dbtlr

**Comment by rickybanister** [comment]
As per my mixed-message comment about the scary red warning notice, that problem could perhaps be alleviated if we used the standard gray notice shown in the very first recommendation. As a warning it should stand out, but it's not a bug or a fatal problem. The light gray might be sufficient.

**Comment by dbtlr** [comment]
I'd like to add the the "Lockdown mode" should contain some sort of confirmation dialog. I don't think anyone should turn this on without understanding what it is going to do to their site. Personally, I hate that something like this is so easy to turn on and would much prefer it to be enabled by defining the constant for WPLOCKDOWN manually. 

Assuming it needs to remain as it is though, I think we need to let the user know exactly what it is going to do to their site and verify that they're okay with that happening.

**Comment by rickybanister** [comment]
We should definitely use the dialog component there. We recently modified that to create a very 'scary' looking disconnection banner for Jetpack. It looks like this:

![image](https://cloud.githubusercontent.com/assets/437258/24804223/bf5e7a88-1b7b-11e7-8eb7-1e7f37dcb374.png)

We could offer some cautionary info in such a dialog.

**Comment by drw158** [comment]
I think I saw the super cache settings once, but I don't remember how to test. Here is some feedback base on the screenshots. Overall, the main tweaks I see are strange wordings.

- I agree "Caching On (recommended)" toggle should be changed. I suggest "Simple caching". "Speed up your site with simple caching" is a bit too wordy for a toggle label. Sounds more like an upgrade notice.
- The whole paragraph that starts with "Mod_rewrite" could use a rewrite. Maybe editorial could help.
- Lockdown: I would make the toggle label even simpler, maybe just "Lockdown" with a description underneath. Maybe a link to a site with more info for devs.
- "Do not cache pages with the following strings" — isn't the string in the textarea a filename? The description could be rewritten. Ex, "Add strings here" not "Add here strings".

**Comment by rickybanister** [comment]
@drw158 to test you go to staging and visit /extensions/wp-super-cache

I tried to keep my feedback here related to patterns only (unless there was an egregiously confusing thing) since the scope of this project was to simply 'port' super cache over. But clearly a scrub of the language would be a great improvement before launch. 

I believe I heard there was a special trigger for getting editorial to review, but I can't remember what it is. @michelleweber am I making that up?

**Comment by michelleweber** [comment]
@rickybanister, you are not making it up :)

(You have activated the trigger by tagging me; you can also tag the whole team (@automattic/editorial) or by adding the "needs copy review" label.)

I'll take a look at this later today, after I finish up some meetup stuff!

**Comment by rickybanister** [comment]
> @automattic/editorial

That's the one!

Thanks again Michelle!

**Comment by michelleweber** [comment]
Agree that "Caching on" is not the best label -- I'm thinking something like "Activate simple caching," to make it a little clearer and keep it in line with how we usually try to word toggles like this.

The PHP para under the toggle (in the first screenshot) could also use a boost:

> PHP caching is enabled, but we also detect Super Cache mod_rewrite rules. Cached files will be served using the mod_rewrite rules. If your site is working properly, you can ignore this message. Otherwise, please edit the .htaccess file in the root of your Super Cache install to remove the mod_rewrite rules.

Super Cache is variously written as "Super Cache," "SuperCache," or "Supercache," so we should be consistent with whichever one is correct -- I'm assuming it's the first.

On the mod_rewrite para -- it feels weird to indent it, as it applies to all three of the sub-options there. Indenting it makes it feel like it only applies to the third option. Also agree that it could use a re-word. Suggestion (bits in parens are things that seem helpful, but I'm not sure they're accurate):

> If you're new to WordPress, we recommend PHP caching, which offers the best combination of speed and ease.  Mod_rewrite is faster (but requires additional configuration), while legacy caching is slower but offers more flexibility (for what?).

Related: I think you could tighten up that Jetpack banner as well: "Disconnecting example.com from WordPress.com removes all access to:"

**Comment by dbtlr** [comment]
The use of the term "simple cache" is a misnomer. 

This toggle under the Easy tab and the toggle with the same name under the Advanced tab both do the same thing, which is turn caching on or off. In fact, they literally do the exact same thing. We can address obvious UX issues with this approach on a later pass. For right now, our directive is to do a UX port as closely as possible from the current plugin then look at these types of oddities separately.

For now, though, I've opted for "Enable Page Caching." #12926

However, I can swap Enable for Activate if that is a better word.

**Comment by michelleweber** [comment]
Noted. Activate/enable is six of one, half a dozen of the other in this instance; enable's fine.
